lib/sensors/stripe_sensor.cpp

```diff
@@ -0,0 +1,26 @@
+#include "stripe_sensor.hpp"
+
+namespace hyped::sensors {
+
+Keyence::Keyence(core::ILogger &log, const std::uint8_t newPin) : pin(newPin), logger_(log)

```

Comment:
as a parameter, take in an `IGpioReader`, then initialise the `gpio_reader_` with it. for this, a `static create` function (like all the IO code) would be good. Its purpose would simply be to create a Keyence object, but the create function itself should take an `IGpio` and a `pin` so it could make a `gpio_reader` to then pass to the constructor of keyence.